vochain/indexer/archive.go

```diff
@@ -0,0 +1,138 @@
+package indexer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.vocdoni.io/dvote/data"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	indexerdb "go.vocdoni.io/dvote/vochain/indexer/db"
+	"go.vocdoni.io/dvote/vochain/indexer/indexertypes"
+	"go.vocdoni.io/dvote/vochain/results"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+const (
+	marxArchiveFileSize     = 1024 * 100 // 100KB
+	timeoutArchiveRetrieval = 60 * time.Second
+	archiveFetchInterval    = 20 * time.Minute
+	archiveFileNameSize     = types.ProcessIDsize * 2 // 64 hex chars
+)
+
+// ArchiveProcess is the struct used to store the process data in the archive.
+type ArchiveProcess struct {
+	ChainID     string                `json:"chainId,omitempty"`
+	ProcessInfo *indexertypes.Process `json:"process"`
+	Results     *results.Results      `json:"results"`
+	StartDate   *time.Time            `json:"startDate,omitempty"`
+	EndDate     *time.Time            `json:"endDate,omitempty"`
+}
+
+// ImportArchive imports an archive list of processes into the indexer database.
+func (idx *Indexer) ImportArchive(archive []*ArchiveProcess) error {
+	tx, err := idx.readWriteDB.Begin()
+	if err != nil {
+		panic(err) // shouldn't happen, use an error return if it ever does
+	}
+	queries := idx.blockQueries.WithTx(tx)
```

Comment:
don't use `blockQueries`, that's only for `Commit` and is protected by `blockMu`. you can always add `importQueries` if you want to keep the prepared queries around for this as well.

Reply:
Can you extend a bit more on that please? What you mean by "add importQueries" ?

Reply:
easiest to not use prepared statements for now. we can always add them later without a data race..